### PR TITLE
2.0.0

### PR DESCRIPTION
**Pull request by rreinold**
Restructured into Library

### CODE REVIEW

File: code/services/AWSS3ExampleFetchAsset/AWSS3ExampleFetchAsset.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/**
+ * Fetch the contents of an access-controlled AWSS3 Resource
+ * 
+ * @param {string} fileName - filename associated with a row in aws_s3_resources collection
+ * @returns {string} content - contents of AWSS3 hosted file
+ */
+function AWSS3ExampleFetchAsset(req, resp) {
+    const COLLECTION = "aws_s3_resources";
+    const PARAMETERIZED = false
```
Comment (yashjain28) [inline code/services/AWSS3ExampleFetchAsset/AWSS3ExampleFetchAsset.js]: Not using this right? const PARAMETERIZED = false

File: system.json [hunk 2]
```diff
@@ -1,10 +1,9 @@
 {
     "auth": true,
-    "deployments": [],
     "description": "",
     "messaging_url": "staging.clearblade.com:1883",
-    "name": "aws-s3-access-control",
+    "name": "packagedevawss3",
```
Comment (yashjain28) [inline system.json]: Not sure what you want to name this? definitely not packagedevawss3 and I'd prefer removing systemkey and secrets.

File: code/services/AWSS3ExampleFetchAsset/AWSS3ExampleFetchAsset.js (hunk 1 above)
Comment (yashjain28) [inline code/services/AWSS3ExampleFetchAsset/AWSS3ExampleFetchAsset.js]: Not using this right? const PARAMETERIZED = false

File: package.json [hunk 3]
```diff
@@ -2,6 +2,12 @@
     "author": {
```
Comment (yashjain28) [inline package.json:2]: I'd usually change it to "Rob Reinold <<rreinold@clearblade.com>>"

File: package.json [hunk 4]
```diff
@@ -2,6 +2,12 @@
...
         "email": "rreinold@clearblade.com"
     },
+    "scripts": {
+        "jsdoc": "jsdoc2md code/libraries/*/*.js > jsdoc.md"
+    },
+    "devDependencies": {
+        "jsdoc-to-markdown": "^4.0.1"
+    },
     "asset_types":["Code Services","Collections"],
```
Comment (yashjain28) [inline package.json]: asset_types = ["Code Libraries", ...]

File: Readme.md [hunk 5]
```diff
@@ -22,35 +22,87 @@ This package contains an example S3 Object to fetch, and can be run upon a fresh
...
 
-Call `FetchS3Object` as an Authenticated User
+___Note___: Make sure to initialize ClearBlade Object, before calling the Library methods. i.e. `ClearBlade.init({request:req});`
 
-### Code Services
+### Code Libraries
 
-`FetchS3Object` - Fetch the contents of an access-controlled AWS S3 Resource
+* `AWSS3Lib` - This library provides a method to get file contents from AWSS3 using a filename. It internally uses a collection to get s3 url for corresponding filename. 
+***Note***: User needs to add a new file, url entry in the collections before trying to access it.
```
Comment (yashjain28) [inline Readme.md]: Wouldn't ask you to do it specially, but as you'll be changing code. 
new `{file, url}` entry in collections. might not be clear when you skim through the lines. 
Again not important

File: Readme.md [hunk 6]
```diff
@@ -22,35 +22,87 @@ This package contains an example S3 Object to fetch, and can be run upon a fresh
...
 ### Collections
 
-`aws_s3_resources` - Stores key/value pairs of filenames and their respective URLs in AWS S3
-### ...
+* `s3_resources` - Stores key/value pairs of filenames and their respective URLs in AWSS3
+
+## API
 
-## Thank you
+## Typedefs
```
Comment (yashjain28) [inline Readme.md]: Not sure what is the problem with jsdoc maybe there is some parameter which generates stuff with ### and not ##
API sections and sub parts of the API section are with `##`

**Review by yashjain28 — CHANGES_REQUESTED** [review]
Small changes

File: code/services/AWSS3ExampleFetchAsset/AWSS3ExampleFetchAsset.js (hunk 1 above)
Comment (rreinold) [reply code/services/AWSS3ExampleFetchAsset/AWSS3ExampleFetchAsset.js]: 4830e9a

File: package.json (hunk 3 above)
Comment (rreinold) [reply package.json:2]: 2154487

File: package.json (hunk 4 above)
Comment (rreinold) [reply package.json]: b2bac4e

File: Readme.md (hunk 5 above)
Comment (rreinold) [reply Readme.md]: 6cc730c

File: Readme.md (hunk 6 above)
Comment (rreinold) [reply Readme.md]: 7738d83

**Pull request merged**